diff --git a/os/os_yield.c b/os/os_yield.c
--- a/os/os_yield.c
+++ b/os/os_yield.c
@@ -10,6 +10,7 @@
 #include "db_config.h"
 
 #include "db_int.h"
+#include <sched.h>
 
 /*
  * __os_yield --
@@ -33,7 +34,7 @@ __os_yield(env, secs, usecs)
 
 #if defined(HAVE_PTHREAD_YIELD)
-	pthread_yield();
+	(void)sched_yield();
 #elif defined(HAVE_SCHED_YIELD)
 	(void)sched_yield();
 #elif defined(HAVE_SELECT)
